zerver/actions/realm_settings.py

```diff
@@ -336,6 +336,9 @@ def do_deactivate_realm(realm: Realm, *, acting_user: Optional[UserProfile]) ->
 
 
 def do_reactivate_realm(realm: Realm) -> None:
+    if not realm.deactivated:
+        return
```

Comment:
Do we want a logging statement here? I'm not sure if warning or info is most appropriate, but we don't expect this to come up in normal operation, and thus it might be helpful to make it easy to investigate?

Reply:
@timabbott Yeah, good idea, pushed with the added logline.